modin/numpy/test/test_array.py

```diff
@@ -0,0 +1,778 @@
+    modin_result = modin_flat_bool_arr.where(x=modin_flat_arr, y=(-1 * modin_flat_arr))
+    numpy_result = numpy.where(
+        numpy_flat_bool_arr, numpy_flat_arr, (-1 * numpy_flat_arr)
+    )
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+    numpy_arr = numpy_flat_arr.reshape((10, 10))
+    modin_arr = np.array(numpy_arr)
+    modin_bool_arr = modin_arr > 0
+    numpy_bool_arr = numpy_arr > 0
+    modin_result = modin_bool_arr.where(modin_arr, 10 * modin_arr)
+    numpy_result = numpy.where(numpy_bool_arr, numpy_arr, 10 * numpy_arr)
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+
+
+def test_max():
+    # Test 1D
+    numpy_arr = numpy.random.randint(-100, 100, size=100)
+    modin_arr = np.array(numpy_arr)
+    assert modin_arr.max() == numpy_arr.max()
+    modin_result = modin_arr.max(axis=0)
+    numpy_result = modin_arr.max(axis=0)
+    assert modin_result == numpy_result
+    modin_result = modin_arr.max(initial=200)
+    numpy_result = numpy_arr.max(initial=200)
+    assert modin_result == numpy_result
+    modin_result = modin_arr.max(initial=0, where=False)
+    numpy_result = numpy_arr.max(initial=0, where=False)
+    assert modin_result == numpy_result
+    modin_result = modin_arr.max(keepdims=True)
+    numpy_result = numpy_arr.max(keepdims=True)
+    assert modin_result.shape == numpy_result.shape
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+    # Test 2D
+    numpy_arr = numpy.random.randint(-100, 100, size=(20, 20))
+    modin_arr = np.array(numpy_arr)
+    assert modin_arr.max() == numpy_arr.max()
+    modin_result = modin_arr.max(axis=0)
+    numpy_result = numpy_arr.max(axis=0)
+    assert modin_result.shape == numpy_result.shape
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+    modin_result = modin_arr.max(axis=0, keepdims=True)
+    numpy_result = numpy_arr.max(axis=0, keepdims=True)
+    assert modin_result.shape == numpy_result.shape
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+    modin_result = modin_arr.max(axis=1)
+    numpy_result = numpy_arr.max(axis=1)
+    assert modin_result.shape == numpy_result.shape
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+    modin_result = modin_arr.max(axis=1, keepdims=True)
+    numpy_result = numpy_arr.max(axis=1, keepdims=True)
+    assert modin_result.shape == numpy_result.shape
+    numpy.testing.assert_array_equal(modin_result._to_numpy(), numpy_result)
+    modin_result = modin_arr.max(initial=200)
+    numpy_result = numpy_arr.max(initial=200)
+    assert modin_result == numpy_result
+    modin_result = modin_arr.max(initial=0, where=False)
+    numpy_result = numpy_arr.max(initial=0, where=False)
+    assert modin_result == numpy_result
+    with pytest.raises(ValueError):
+        modin_result = modin_arr.max(out=modin_arr, keepdims=True)
```

Comment:
codeql warning here, i'm assuming this is just checking for the presence of an error, but if that's the case we should remove the assign to `modin_result`

Reply:
Makes sense - will fix!